src/transports/dash/manifest_parser.ts

```diff
@@ -177,6 +177,11 @@ export default function generateManifestParser(
           });
           return processMpdParserResponse(value.continue(resources));
         }
+      }).catch(() => {
+        if (value.format === "string") {
```

Comment:
I'm not a fan of this solution, because there could be legitimate cases where a fetched resource would return an empty string.

To me, it would be better to be very explicit that we failed the request:

  - we could continue with a whole other response format (for example, a discriminated union (like `{ success: false, error: "GNAGNA" } | { success: true, response: data  }`)
  
  - we could add another callback for example `continue` (or a renamed one) on success or `onFailure` on failure
  
  -  we could do it nodejs-style, by adding a second argument somewhere (in nodejs API cases, there's often an `error` argument that would be `null` when no error happened)

I'm sure there could be other ways.